openmsupply/open-msupply
# 401 program event table

- Adds program event table (this table is reasonable generic enough to be used for other purposes in the future)
- Service + API to update events and to query them
- DispensedDuration Control to schedule events based on the control's "number of days dispensed" value

Testing:
- reinit the DB
- Program for Tina will show status "Lost to follow up"
- Go to one of her encounters and set the "Number of  Days Dispensed" field to a value > 0 (there are two controls for it use the top one)
- Save the encounter
- Status should now be cleared
- wait 1 minute -> status change to "Interupted", wait another 2 minutes -> status changes to "Lost to follow up"

Some program event table motivation:
- Events have a patient + context which can be used to assign them, for example, to a program enrolment or some other entity 
- Within this context events can be grouped (group field). This is useful to replace a group of events with an updated list of events in this group, leaving other events in the context untouched.

Closes #401
